### PR TITLE
Adds supports for unicode-encoded names when creating an avatar from a name's initial

### PR DESCRIPTION
**Pull request by StCyr**
Fix to https://github.com/nextcloud/nextcloud-vue/issues/1406

not tested though :-)

Signed-off-by: Cyrille Bollu <cyrpub@bollu.be>

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
Bildschirmfoto von 2020-10-13 11-12-38

Does not seem to work as expected?

**Comment by ChristophWurst** [comment]
Use this 
Bildschirmfoto von 2020-10-13 11-15-43

Ref https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt

**Comment by StCyr** [comment]
> Use this
> Bildschirmfoto von 2020-10-13 11-15-43
> 
> Ref https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt
> 
> And it also work with simple names
> 
> Bildschirmfoto von 2020-10-13 11-16-29

Thanks! I'll update my PR (sorry for the wrong solution I've proposed)

**Comment by skjnldsv** [comment]
Btw shouldn't we use the double initials for the displayname?

**Comment by StCyr** [comment]
> Btw shouldn't we use the double initials for the displayname?

I also wonder.

I think it was the case earlier, and then, at some point in time, NC has started to display only 1 initial...

**Comment by skjnldsv** [comment]
> I think it was the case earlier, and then, at some point in time, NC has started to display only 1 initial...

It's the opposite, we showed only one, but now show two :wink: 
https://github.com/nextcloud/server/pull/18745

**Comment by StCyr** [comment]
> > I think it was the case earlier, and then, at some point in time, NC has started to display only 1 initial...
> 
> It's the opposite, we showed only one, but now show two 😉
> [nextcloud/server#18745](https://github.com/nextcloud/server/pull/18745)
> https://github.com/nextcloud/server/blob/caff1023ea72bb2ea94130e18a2a6e2ccf819e5f/lib/private/Avatar/Avatar.php#L98-L100

Ha! nice 👍 

I can push this change here if you want, that would be great I think

**Comment by skjnldsv** [comment]
> I can push this change here if you want, that would be great I think

Sure, it feels relevant enough! :+1:

File: src/components/Avatar/Avatar.vue [hunk 1]
```diff
@@ -314,10 +314,20 @@ export default {
 			return this.displayName
 		},
 		initials() {
+			let initials
 			if (this.shouldShowPlaceholder) {
-				return this.getUserIdentifier.charAt(0).toUpperCase()
+				const userId = this.getUserIdentifier
```
Comment (skjnldsv) [inline src/components/Avatar/Avatar.vue:319]: please use something else than `userId`, this is also the displayName :)

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**